File: prakash/15_ArrayIntersection/ArrayIntersection.test.js
const arrayIntersection = require("./ArrayIntersection");

describe("Array Intersection", () => {
  test("Test with arrays having common elements", () => {
    const nums1 = [1, 2, 2, 1];
    const nums2 = [2, 2, 3];
    const result = arrayIntersection(nums1, nums2);
    expect(result).toEqual([2]);
  });

  test("Test with arrays having no common elements", () => {
    const nums1 = [1, 2, 3];
    const nums2 = [4, 5, 6];
    const result = arrayIntersection(nums1, nums2);
    expect(result).toEqual([]);
  });

  test("Test with empty arrays", () => {
    const nums1 = [];
    const nums2 = [];
    const result = arrayIntersection(nums1, nums2);
    expect(result).toEqual([]);
  });

  test("Test with arrays containing duplicate elements", () => {
    const nums1 = [4, 9, 5, 9];
    const nums2 = [9, 4, 9, 8, 4];
    const result = arrayIntersection(nums1, nums2);
    expect(result.sort()).toEqual([4, 9].sort()); // Sort both arrays
  });
});

File: Bipin/015_array_intersection/array_intersection_test.go
package main

import (
	"reflect"
	"testing"
)

func TestIntersection(t *testing.T) {
	// Test 1: Arrays with common elements
	nums1 := []int{1, 2, 2, 1}
	nums2 := []int{2, 2, 3}
	expected := []int{2}
	result := Intersection(nums1, nums2)
	if !reflect.DeepEqual(result, expected) {
		t.Errorf("Test 1 failed. Expected %v, got %v", expected, result)
	}

	// Test 2: Arrays with no common elements
	nums1 = []int{4, 5, 6}
	nums2 = []int{7, 8, 9}
	expected = []int{}
	result = Intersection(nums1, nums2)
	if !reflect.DeepEqual(result, expected) {
		t.Errorf("Test 2 failed. Expected %v, got %v", expected, result)
	}

	// Test 3: Both arrays are empty
	nums1 = []int{}
	nums2 = []int{}
	expected = []int{}
	result = Intersection(nums1, nums2)
	if !reflect.DeepEqual(result, expected) {
		t.Errorf("Test 3 failed. Expected %v, got %v", expected, result)
	}

	// Test 4: Arrays with duplicates
	nums1 = []int{1, 1, 2, 2}
	nums2 = []int{2, 2, 1, 1}
	expected = []int{2, 1}
	result = Intersection(nums1, nums2)
	if !reflect.DeepEqual(result, expected) {
		t.Errorf("Test 4 failed. Expected %v, got %v", expected, result)
	}
}
